app/src/main/java/com/example/android/unsplash/MainActivity.java

```diff
@@ -89,22 +91,35 @@ private void displayData() {
         if (relevantPhotos != null) {
             populateGrid();
         } else {
-            UnsplashService unsplashApi = new RestAdapter.Builder()
-                    .setEndpoint(UnsplashService.ENDPOINT)
-                    .build()
-                    .create(UnsplashService.class);
-            unsplashApi.getFeed(new Callback<List<Photo>>() {
+            OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
+
+            Retrofit.Builder builder =
+                    new Retrofit.Builder()
+                            .baseUrl(UnsplashService.ENDPOINT)
+                            .addConverterFactory(
+                                    GsonConverterFactory.create()
+                            );
+
+            Retrofit retrofit =
+                    builder
+                            .client(
+                                    httpClient.build()
+                            )
+                            .build();
+
+            UnsplashService unsplashApi =  retrofit.create(UnsplashService.class);
+            unsplashApi.getFeed().enqueue(new Callback<List<Photo>>() {
                 @Override
-                public void success(List<Photo> photos, Response response) {
+                public void onResponse(Call<List<Photo>> call, Response<List<Photo>> response) {
                     // the first items not interesting to us, get the last <n>
-                    relevantPhotos = new ArrayList<>(photos.subList(photos.size() - PHOTO_COUNT,
-                            photos.size()));
+                    relevantPhotos = new ArrayList<>(response.body().subList(response.body().size() - PHOTO_COUNT,
```

Comment:
Hold on to photos for the duration of the response. Makes it easier to read than multiple `response.body()` calls.